src/pseudopeople/interface.py

```diff
@@ -30,64 +31,70 @@ def _generate_form(
         Noised form data in a pd.DataFrame
     """
     configuration_tree = get_configuration(configuration)
+    # TODO: we should save outputs of the simulation with filenames that are
+    #  consistent with the names of the forms if possible.
+    form_file_name = {
+        FORMS.acs.name: "household_survey_observer_acs",
+        FORMS.cps.name: "household_survey_observer_cps",
+        FORMS.tax_w2_1099.name: "tax_w2_observer",
+        FORMS.wic.name: "wic_observer",
+    }.get(form.name, f"{form.name}_observer")
     if source is None:
-        # TODO: hard-coding the .parquet extension for now. This will go away
-        #  once we only support passing the root directory of the data.
-        # TODO: we should save outputs of the simulation with filenames that are
-        #  consistent with the names of the forms if possible.
-        form_file_name = {
-            FORMS.acs.name: "household_survey_observer_acs",
-            FORMS.cps.name: "household_survey_observer_cps",
-            FORMS.tax_w2_1099.name: "tax_w2_observer",
-            FORMS.wic.name: "wic_observer",
-        }.get(form.name, f"{form.name}_observer")
-
-        source = paths.SAMPLE_DATA_ROOT / form_file_name / f"{form_file_name}.parquet"
-    if isinstance(source, str):
-        source = Path(source)
-    if isinstance(source, pd.DataFrame):
-        data = source
-    elif isinstance(source, Path):
-        if source.suffix == ".hdf":
-            data = pd.read_hdf(source)
-        elif source.suffix == ".parquet":
-            data = pd.read_parquet(source)
+        source = paths.SAMPLE_DATA_ROOT
+    source = Path(source) / form_file_name
+    data_paths = [x for x in source.glob(f"{form_file_name}*")]
+    if not data_paths:
+        logger.warning(
+            f"No datasets found at directory {str(source)}. "
+            "Please provide the path to the unmodified root data directory."
+        )
+        return None
+    suffix = set(x.suffix for x in data_paths)
+    if len(suffix) > 1:
+        raise TypeError(
+            f"Only one type of file extension expected but more than one found: {suffix}. "
+            "Please provide the path to the unmodified root data directory."
+        )
+    noised_form = []
+    for data_path in data_paths:
+        if data_path.suffix == ".hdf":
+            data = pd.read_hdf(data_path)
+        elif data_path.suffix == ".parquet":
+            data = pd.read_parquet(data_path)
```

Comment:
This function is becoming spaghetti code.

Reply:
Yeah I actually had it as a utility function but then realized I only had it here. But then I updated the tests and it's used there as well. I could go either way